internal/cli/run_test.go

```diff
@@ -1225,6 +1225,260 @@ var _ = Describe("Run", func() {
 			})
 		})
 
+		Context("when quarantined tests should not be retried", func() {
```

Comment:
FWIW these tests rely on the mocked parser (because all tests in this file mock the parser), which means that as long as the _mock parser_ filters out the quarantined test, this test would pass (even if the application code does not filter it out).

For that reason I've also extracted the filter function into its own public function for more isolated testing, which tests that it filters out quarantined tests, but can't ensure that they actually get skipped by the code that calls the filter function, so I think it makes sense to test at both levels.